flexmeasures/data/models/planning/linear_optimization.py

```diff
@@ -339,10 +339,16 @@ def cost_function(m):
     model.costs = Objective(rule=cost_function, sense=minimize)
 
     # Solve
+
+    # load_solutions=False to avoid a RuntimeError exception in appsi solvers when solving an infeasible problem.
     results = SolverFactory(current_app.config.get("FLEXMEASURES_LP_SOLVER")).solve(
-        model
+        model, load_solutions=False
     )
 
+    # load the results only if the termination conditions is not infeasible
+    if results.solver.termination_condition != TerminationCondition.infeasible:
```

Comment:
It looks like checking the length of `results.solution` is going to be more robust, because there are some other termination conditions that would result in the same `RuntimeError('A feasible solution was not found [...]`. Checking the length of the solutions list is also done elsewhere within Pyomo, e.g. [here](https://github.com/ZedongPeng/pyomo/blob/ba15736791e0ef944cc62f705914b23b28f55c86/pyomo/contrib/mindtpy/mip_solve.py#L70).

```
# load the results only if a feasible solution has been found
if len(results.solution) > 0:
```

Reply:
Thanks for checking it out!